modules/gapi/include/opencv2/gapi/infer/ie.hpp

```diff
@@ -88,6 +88,15 @@ struct ParamDesc {
 
     cv::optional<cv::gapi::wip::onevpl::Device> vpl_preproc_device;
     cv::optional<cv::gapi::wip::onevpl::Context> vpl_preproc_ctx;
+
+    using precision_t = int;
+    using precision_map_t = std::unordered_map<std::string, int>;
+    // NB: cv::util::monostate is default value that means precision wasn't specified.
+    using precision_variant_t = cv::util::variant<cv::util::monostate,
+                                                  precision_t,
+                                                  precision_map_t>;
```

Comment:
Can you please document the cases in this variant?

My understanding (I haven't read the rest of the patch yet) is that `precision_t` case is for single output layer, right? And map is for multi-output case?

Reply:
No, user can do:
```
params.cfgOutputPrecision(`CV_16F`) // will be applied for all layers.
```
or specify certain layers:
```
params.cfgOutputPrecision({{"layer0", CV_16F}}); // only layer0 has changed precision 
```

Reply:
Done

Reply:
```
// will be applied for all layers
```
Well I failed to guess that. :)